sdk/nodejs/cmd/pulumi-language-nodejs/main.go

```diff
@@ -1345,7 +1353,86 @@ func (host *nodeLanguageHost) GetProgramDependencies(
 func (host *nodeLanguageHost) RunPlugin(
 	req *pulumirpc.RunPluginRequest, server pulumirpc.LanguageRuntime_RunPluginServer,
 ) error {
-	return errors.New("not supported")
+	logging.V(5).Infof("Attempting to run nodejs plugin in %s", req.Info.ProgramDirectory)
+	ctx := context.Background()
+
+	closer, stdout, stderr, err := rpcutil.MakeRunPluginStreams(server, false)
+	if err != nil {
+		return err
+	}
+	// best effort close, but we try an explicit close and error check at the end as well
+	defer closer.Close()
+
+	nodeBin, err := exec.LookPath("node")
+	if err != nil {
+		return err
+	}
+
+	if logging.V(5) {
+		commandStr := strings.Join(req.Args, " ")
+		logging.V(5).Infoln("Language host launching process: ", nodeBin, commandStr)
+	}
+
+	opts, err := parseOptions(req.Info.Options.AsMap())
+	if err != nil {
+		return err
+	}
+
+	env := os.Environ()
+	if opts.typescript {
+		env = append(env, "PULUMI_NODEJS_TYPESCRIPT=true")
+	}
+	if opts.tsconfigpath != "" {
+		env = append(env, "PULUMI_NODEJS_TSCONFIG_PATH="+opts.tsconfigpath)
+	}
+
+	runPath := os.Getenv("PULUMI_LANGUAGE_NODEJS_RUN_PATH")
+	if runPath == "" {
+		runPath = defaultRunPluginPath
+	}
+
+	runPath, err = locateModule(ctx, runPath, req.Info.ProgramDirectory, nodeBin, true)
+	if err != nil {
+		return err
+	}
+
+	args := []string{runPath}
+
+	nodeargs, err := shlex.Split(opts.nodeargs)
+	if err != nil {
+		return err
+	}
+
+	nodeargs = append(nodeargs, req.Info.EntryPoint)
+
+	args = append(args, nodeargs...)
+
+	// Now simply spawn a process to execute the requested program, wiring up stdout/stderr directly.
+	cmd := exec.Command(nodeBin, args...) // nolint: gas // intentionally running dynamic program name.
```

Comment:
Does that `nolint` work? I thought they had to be `//nolint` with no spaces.

Reply:
The linter complains about the spaces as well.  It looks like this nolint is not necessary at all though, so I'll just remove it.